src/veh_type.h

```diff
@@ -127,6 +133,33 @@ class vpart_info
         /** Mechanics skill required to install item */
         int difficulty = 0;
 
+        /** Legacy parts don't specify installation requirements */
+        bool legacy = true;
+
+        /** Installation requirements for this component */
+        requirement_data install_reqs;
+
+        /** Required skills to install this component */
+        std::map<skill_id, int> install_skills;
+
+        /** Installation time (in moves) for component (@see install_time), default 1 hour */
+        int install_moves = MOVES( HOURS( 1 ) );
+
+        /** Installation time (in moves) for this component accounting for player skills */
+        int install_time( const Character &ch ) const;
+
+        /** Requirements for removal of this component */
+        requirement_data removal_reqs;
+
+        /** Required skills to remove this component */
+        std::map<skill_id, int> removal_skills;
+
+        /** Removal time (in moves) for component (@see removal_time), default is half @ref install_moves */
+        int removal_moves = -1;
+
+        /** Removal time (in moves) for this component accounting for player skills */
+        int removal_time( const Character &ch ) const;
```

Comment:
I also want to add repair requirements so I'm going to undertake the above consolidation then once the design is finalised.